src/internal/connector/exchange/service_iterators.go

```diff
@@ -162,50 +163,42 @@ func FetchEventIDsFromCalendar(
 	gs graph.Service,
 	user, calendarID string,
 ) ([]string, error) {
-	ids := []string{}
+	var (
+		errs *multierror.Error
+		ids  []string
+	)
 
-	response, err := gs.Client().
+	query := gs.Client().
 		UsersById(user).
 		CalendarsById(calendarID).
-		Events().Get(ctx, nil)
-	if err != nil {
-		return nil, errors.Wrap(err, support.ConnectorStackErrorTrace(err))
-	}
+		Events()
 
-	pageIterator, err := msgraphgocore.NewPageIterator(
-		response,
-		gs.Adapter(),
-		models.CreateEventCollectionResponseFromDiscriminatorValue,
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "iterator creation failure during fetchEventIDs")
-	}
+	for {
+		resp, err := query.Get(ctx, nil)
```

Comment:
should be able to use an option here to minimize the amount of data being pulled.